backend/pkg/models/user.go

```diff
@@ -4,21 +4,19 @@ import (
 	"fmt"
 	"strings"
 
-	"golang.org/x/crypto/bcrypt"
-
 	"github.com/fastenhealth/fasten-onprem/backend/pkg"
+	"golang.org/x/crypto/bcrypt"
 )
 
 type User struct {
 	ModelBase
-	FullName string `json:"full_name"`
-	Username string `json:"username" gorm:"unique"`
-	Password string `json:"password"`
-
-	//additional optional metadata that Fasten stores with users
-	Picture string       `json:"picture"`
-	Email   string       `json:"email"`
-	Role    pkg.UserRole `json:"role"`
+	FullName    string                     `json:"full_name"`
+	Username    string                     `json:"username" gorm:"unique"`
+	Password    string                     `json:"password"`
+	Picture     string                     `json:"picture"`
+	Email       string                     `json:"email"`
+	Role        pkg.UserRole               `json:"role"`
+	Permissions map[string]map[string]bool `json:"permissions" gorm:"-:all"`
```

Comment:
can we use GORM references for this - https://gorm.io/docs/has_many.html#Override-References